src/MiNET/MiNET/Player.cs

```diff
@@ -200,16 +206,32 @@ public virtual void HandleMcpePhotoTransfer(McpePhotoTransfer message)
 
 		protected Form CurrentForm { get; set; } = null;
 
+		public virtual void SendForm(Form form)
+		{
+			CurrentForm = form;
+
+			McpeModalFormRequest message = McpeModalFormRequest.CreateObject();
+			message.formId = form.Id; // whatever
+			message.data = form.ToJson();
+			SendPacket(message);
+		}
+
 		public void HandleMcpeModalFormResponse(McpeModalFormResponse message)
 		{
 			if (CurrentForm == null) Log.Warn("No current form set for player when processing response");
+			if (message.cancelReason == 1)
```

Comment:
Can we add an enum with cancelReasons? I know these to be as follows:
UserClosed = 0
UserBusy = 1